main/test/flix/Test.Exp.Regions.flix

```diff
@@ -298,10 +302,10 @@ namespace Test/Exp/Regions {
     @test
     def testRegions2Regions04(): Bool \ IO =
         region r1 {
-            let a1 = [1; 8] @ r1;
+            let a1 = $ARRAY_REPEAT$(r1, 1, 8);
             a1[0] == 1 and 1 ==
             region _ {
-                let a2 = [1; 8] @ Static;
+                let a2 = $ARRAY_REPEAT$(unsafe_cast magic() as Region[false], 1, 8);
```

Comment:
This file is supposed to test regions, not all kinds of ARRAY_REPEAT, unsafe cast, and magic stuff! :)

Reply:
Yeah but our safety rules are sort of ruining things here / making things complicated.

$ARRAY_REPEAT$ requires a term level value that carries region information. So naturally you would unsafe_cast unit to some region type but that is rejected as a safety error. Then we have to introduce a function whose return type is a type variable so the compiler accepts an unsafe_cast from a flix primitive type to a type variable.

This is all just to allocate memory in the static region.

Reply:
All just to do the equivalent of

[1;8] @ Static

Also, $ARRAY_REPEAT$(false, ...) doesn't work as false is a Boolean not a region (just stating the obvious here)

Reply:
Can we just use a normal region?

Reply:
We are trying to get rid of static anyway

Reply:
Ah perfect! That's much easier haha